pkg/runtime/function2/cast.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package function2
+
+import (
+	"context"
+)
+
+import (
+	"github.com/arana-db/arana/pkg/proto"
+
+	"github.com/pkg/errors"
+)
+
+const FuncCast = "CAST"
+
+var _ proto.Func = (*castFunc)(nil)
+
+func init() {
+	proto.RegisterFunc(FuncCast, castFunc{})
+}
+
+type castFunc struct{}
+
+func (c castFunc) Apply(ctx context.Context, inputs ...proto.Valuer) (proto.Value, error) {
+	val, err := inputs[0].Value(ctx)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	signCast := func(i uint) *int {
+		var sign int
+		return &sign
+	}
```

Comment:
Too simple, and no need to return `*int`. 

Please consider these cases:

1. CAST(3.14 AS SIGNED)
2. CAST(3.14 AS UNSIGNED)
3. CAST(3.14 AS CHAR)
4. CAST(3.14 AS NCHAR)
5. ... (TBD: mysql8.0+)

So I think it should accept 2 arguments, first is the input value to be cast, second is the target type in string, for example:
```go
myCastFunc.Apply(ctx, 3.14, "CHAR")
```

PS: `CONVERT` has similar behavior.

Reply:
> Too simple, and no need to return `*int`. 
> 
> 
> 
> Please consider these cases:
> 
> 
> 
> 1. CAST(3.14 AS SIGNED)
> 
> 2. CAST(3.14 AS UNSIGNED)
> 
> 3. CAST(3.14 AS CHAR)
> 
> 4. CAST(3.14 AS NCHAR)
> 
> 5. ... (TBD: mysql8.0+)
> 
> 
> 
> So I think it should accept 2 arguments, first is the input value to be cast, second is the target type in string, for example:
> 
> ```go
> 
> myCastFunc.Apply(ctx, 3.14, "CHAR")
> 
> ```
> 
> 
> 
> PS: `CONVERT` has similar behavior.

Sure I will fix those. But this pull request just for `CAST` function placeholder,not a complete implementation, for other `CAST` function developers to use this file for entrypoint.